connection/connection.go

```diff
@@ -89,21 +97,28 @@ func ExitOnConnectionLoss() func() bool {
 
 type options struct {
 	reconnect func() bool
+	timeout   time.Duration
 }
 
 // connect is the internal implementation of Connect. It has more options to enable testing.
-func connect(address string, dialOptions []grpc.DialOption, connectOptions []Option) (*grpc.ClientConn, error) {
```

Comment:
Decided to remove this because I don't think it makes sense to pass in dial options into an unexported function.